src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTagger.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
+{
+    /// <summary>
+    /// The purpose of this tagger is to convert the <see cref="InlineParameterNameHintDataTag"/> to
+    /// the <see cref="InlineParameterNameHintsTag"/>, which actually creates the UIElement. It reacts to
+    /// tags changing and updates the adornments accordingly.
+    /// </summary>
+    internal sealed class InlineParameterNameHintsTagger : ITagger<IntraTextAdornmentTag>, IDisposable
```

Comment:
@jasonmalinowski @akhera99 should this be an IAccurateTagger as well?  if not, why not?

Reply:
@CyrusNajmabadi I believe Jason and I discussed this and, correct me if I am wrong, the purpose of implementing the IAccurateTagger is for thread safety concerns? I think since the tagger and the UI element are being created on a completely separate thread than that of the tagger provider which is finding the locations of the adornments means it's going to be threadsafe inherently?

Reply:
actually scratch this.  feels unlikely we would ever need an iaccuratetagger here.